addons/api/addon/serializers/storage-bucket.js

```diff
@@ -12,27 +12,49 @@ const fieldsByCredentialType = {
     'secret_access_key',
     'region',
     'disable_credential_rotation',
+    'endpoint_url',
   ],
   dynamic: [
     'role_arn',
     'role_external_id',
     'role_session_name',
     'role_tags',
-    'disable_credential_rotation',
     'region',
+    'disable_credential_rotation',
+    'endpoint_url',
   ],
 };
 export default class StorageBucketSerializer extends ApplicationSerializer {
   serialize(snapshot) {
-    const serialized = super.serialize(...arguments);
     const { credentialType } = snapshot.record;
+    const pluginType = snapshot.record.plugin?.name;
+    const serialized = super.serialize(...arguments);
```

Comment:
I am trying to follow [credential-store](https://github.com/hashicorp/boundary-ui/blob/main/addons/api/addon/serializers/credential-store.js) approach, but since in this case there are some common serialization I decided to serialize here, and pass to specific type methods the already serialized object. Let me know if I should follow a different approach.

Reply:
I think we should refactor and simplify this serializer if we introduce `for` in the generated model, take a look at `authmethod` [model](https://github.com/hashicorp/boundary-ui/blob/main/addons/api/addon/generated/models/auth-method.js#L214).

Reply:
@DhariniJeeva 
After a conversation on slack, the storage bucket can not be serialized using the `for` to map with a type. The type in storage bucket is only plugin, so we can not use it to differentiate between plugin types, therefor the usage of `for` at serializing will not work. 
In the storage-bucket model, we are using `compositeType` to identify the plugin type.